state_machine/applications/system-check/tests/logging_test.py

```diff
@@ -0,0 +1,84 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+SD Card Logging Test
+* Author(s): Yashika Batra
+"""
+
+import os
+
+
+def sd_test():
+    """
+    Run a basic test; Create a file and test existence, write to and
+    read from a file, delete file and test existence. Return result_dict
+    values accordingly
+    """
+
+    # create filepaths
+    filepath = "test.txt"
+    filepath_directory = "/sd/test.txt"
+
+    # try to create a file with the test filepath
+    try:
+        test_file = open(filepath_directory, "x")
+    except OSError:
+        # if the file already exists, remove it and create a new file
+        os.remove(filepath_directory)
+        test_file = open(filepath_directory, "x")
+
+    print(f"Directory after test.txt was created:\n{os.listdir('/sd/')}")
+    if filepath not in os.listdir("/sd/"):
+        return ("File creation failed.", False)
+
+    # write to file
+    test_string = "Hello World! This is a test file.\n"
+    try:
+        test_file.write(test_string)
+    except OSError as e:
+        print(f"Unable to write to file. {e}")
+        return (f"Unable to write to file. {e}", False)
+    test_file.close()
+
+    # read from file
+    try:
+        test_file_read = open(filepath_directory, "r")
+        test_string_read = test_file_read.read()
+    except OSError as e:
+        print(f"Unable to read from file. {e}")
+        return (f"Unable to read from file. {e}", False)
+
+    if test_string_read != test_string:
+        print("File not written to or read from correctly.")
+        return ("File not written to or read from correctly.", False)
+
+    # delete file
+    os.remove(filepath_directory)
+    print(f"Directory after test.txt was removed:\n{os.listdir('/sd/')}")
+
+    if filepath in os.listdir("/sd/"):
+        return ("File deletion failed.", False)
+
+    # if nothing has failed so far, return success
+    return ("SD Card passed all tests: New file was created, wrote to, " +
+            "read from, and deleted successfully.", True)
+
+
+def run(hardware_dict, result_dict):
+    """
+    Check that the correct hardware is initialized and run tests
```

Comment:
Would be good to be more descriptive in your docstring here.

Reply:
Maybe "Run SD card tests if the SD card was properly initialized, updating result_dict with the test results".